provider/provider.go

```diff
@@ -90,8 +90,14 @@ func (p *dockerNativeProvider) Configure(_ context.Context, req *rpc.ConfigureRe
 		return nil, err
 	}
 	host := client.DaemonHost()
-	log.Printf("Setting DOCKER_HOST to %s", host)
-	os.Setenv("DOCKER_HOST", host)
+
+	// In the case of a remote docker client that we connect to with SSH, we use a connection helper as the daemon host.
+	// Per the docker/cli/cli/connhelper package, this is hardcoded as http://docker.example.com for SSH.
+	// If we are using a remote host via SSH, we do NOT want to overwrite DOCKER_HOST here.
+	if host != "http://docker.example.com" {
+		log.Printf("Setting DOCKER_HOST to %s", host)
+		os.Setenv("DOCKER_HOST", host)
```

Comment:
Both before and after the change this is using os.Setenv that has process scope. Is it at all possible to avoid doing os.Setenv and pass the configuration via an API? Or is this necessary?

Reply:
I'm not sure - as I understand it, this option turns up if
- we do not set the host explicitly (i.e. it is passed via env vars)
- if the default host does not succeed, we try the user host
- we set the env var to the host that works, which in the case of an explicitly configured host (passed in via confiv not env) would be the same as in the config file.

I'm wondering if we can set `p.config["host"]` instead? 🤔